src/serialization/CCLACSets.jl

```diff
@@ -0,0 +1,180 @@
+    dict = Dict{String,AnElement}()
+    foreach(json.notebook.cells) do cell
+        @match cell.content.tag begin
+            "object" => begin
+                type = subpart(predecessor, incident(predecessor, cell.content[:over][:content], :vUuid), :vlabel)
+                push!(dict, cell.content[:id] => ObjectElem(cell.content[:name], only(type)))
+            end
+            "morphism" => begin
+                name = if isempty(cell.content[:name])
+                    if isnothing(cell.content[:over])
+                        cell.content[:name]
+                    else
+                        cell.content[:over][:content]
+                    end
+                else
+                    cell.content[:name]
+                end
+                # TODO elabel is empty because it is a scalar
+                label = @something emptyMaybe(subpart(predecessor, incident(predecessor, name, :eUuid), :elabel)) emptyMaybe(name) ["EMPTY"]
+                if label == ["EMPTY"]
+                    @warn cell
+                end
+                push!(dict, cell.content[:id] => HomElem(only(label), cell.content[:dom][:content], cell.content[:cod][:content]))
+            end
+        end
+    end
+    new(gensym(), dict)
+end
+end
+export CCLDiagram
+
+# Interface
+###########
+
+struct ImplError <: Exception
+    type::String
+end
+
+Base.showerror(io, e::ImplError) = println(io, "Parsing for `$(e.type)` not yet implemented")
+
+"""
+Dispatches parser based on value in `type` field
+"""
+function parse_json(json::AbstractDict; predecessor=nothing)
+    @match json.type begin
+        "model" => CCLModel(json, predecessor)
+        "diagram" => CCLDiagram(json, predecessor)
+        x => throw(ImplError(json.type))
+    end
+end
+
+SchLabeledGraph = BasicSchema([:E, :V], 
+                              [(:src, :E, :V), (:tgt, :E, :V)], [:Label, :UUID], 
+                              [(:vlabel, :V, :Label), (:elabel, :E, :Label), (:vtype, :V, :Label)
+             ,(:vUuid, :V, :UUID), (:eUuid, :E, :UUID)])
+@acset_type UUIDLabeledGraph(SchLabeledGraph, index=[:src,:tgt])
+
+emptyMaybe(x) = isempty(x) ? nothing : Some(x)
+emptyMaybe(x::Union{RegexMatch, Nothing}) = isnothing(x) ? nothing : Some(x)
+
+@active Re{r :: Regex}(x) begin emptyMaybe(match(r, x)); end
+
+function UUIDLabeledGraph(m::CCLModel)
+    g = UUIDLabeledGraph{String, String}()
+    # TODO a little redundant
+    newobs = Dict()
+    newhoms = Dict()
+    foreach(pairs(m.elems)) do (k,v)
+        @match v begin
+            ::ObjectElem => push!(newobs, k => v)
+            ::HomElem => push!(newhoms, k => v)
+        end
+    end
+    nc = 0
+    foreach(pairs(newobs)) do (k,v) 
+        typelabel = @match v.name begin
+            Re{r"Primal (\d)-form"}(x) => "Form$(first(x.captures))"
+            Re{r"Dual (\d)-form"}(x) => "DualForm$(first(x.captures))"
+            Re{r"(\d)-form"}(x) => "Form$(first(x.captures))"
+            x => x
```

Comment:
I'm surprised to see Decapodes-specific logic here: I would have thought this graph would be independent of the Decapodes application.